includes/Handlers/Connection.php

```diff
@@ -1032,4 +1083,260 @@ public function get_plugin() {
 	}
 
 
+	/**
+	 * Process WebHook User object, install field
+	 *
+	 * @link https://developers.facebook.com/docs/marketing-api/fbe/fbe2/guides/get-features#webhook
+	 *
+	 * @param object $data WebHook event data.
+	 */
+	public function fbe_install_webhook( $data ) {
+
+		// Reject other objects other than subscribed object
+		if ( empty( $data ) || ! isset( $data->object ) || self::WEBHOOK_SUBSCRIBED_OBJECT !== $data->object ) {
+
+			if ( $this->get_plugin()->get_integration()->is_debug_mode_enabled() ) {
+				$this->get_plugin()->log( 'Wrong (or empty) WebHook Event received' );
+				$this->get_plugin()->log( print_r( $data, true ) ); //phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_print_r
+			}
+
+			return;
+		}
+
+		$log_data = array();
+		if ( $this->get_plugin()->get_integration()->is_debug_mode_enabled() ) {
+			$this->get_plugin()->log( 'WebHook User Event received' );
+			$this->get_plugin()->log( print_r( $data, true ) ); //phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_print_r
+		}
+
+		$entry = (array) $data->entry[0];
+		if ( empty( $entry ) ) {
+			return;
+		}
+
+		// Filter event by subscribed field
+		$event = array_filter(
+			$entry['changes'],
+			function( $change ) {
+				return self::WEBHOOK_SUBSCRIBED_FIELD === $change->field;
+			}
+		);
+
+		$values = ! empty( $event[0] ) ? $event[0]->value : '';
+
+		/**
+		 * If profiles, pages and instagram_profiles fields are not included in the Webhook payload, this means the business has uninstalled FBE.
+		 * In this case also the field access_token will not be included.
+		 *
+		 * @link https://developers.facebook.com/docs/marketing-api/fbe/fbe2/guides/get-features#what-s-included-with-webhooks-
+		 */
+		if ( empty( $values->access_token ) ) {
+
+			delete_option( 'wc_facebook_has_connected_fbe_2' );
+			delete_option( 'wc_facebook_has_authorized_pages_read_engagement' );
+
+			$this->disconnect();
+
+			return;
+		}
+
+		update_option( 'wc_facebook_has_connected_fbe_2', 'yes' );
+		update_option( 'wc_facebook_has_authorized_pages_read_engagement', 'yes' );
+
+		$system_user_access_token = ! empty( $values->access_token ) ? sanitize_text_field( $values->access_token ) : '';
+		$this->update_access_token( $system_user_access_token );
+		$log_data[ self::OPTION_ACCESS_TOKEN ] = $system_user_access_token;
```

Comment:
Is this sensitive data ending up in the logs? Would this require masking?

Reply:
Thanks, I will change that for a simple: _Token was saved_. This is only [in case debug mode is on](https://github.com/saucal/facebook-for-woocommerce/pull/12/files#diff-b3619b45c706deb9961d595d2adad8e3466bfe5ce4c0e4f451deba104bdc5d7eR1225) tho.